### PR TITLE
[REPL] Rewrite dictionary testing & improve infra to be more reliable.

### PR DESCRIPTION
**Pull request by dcci**
This now uses lit. It's shorter and hopefully more reliable than
pexpect. As part of this, I also did all the plumbing for testing
the repl in lldb, so what I've done here can be used for other repl
tests. While I'm at it, remove the old test, we don't need it anymore.

<rdar://problem/37407231>

### CODE REVIEW

**Comment by dcci** [comment]
@swift-ci please test

File: lit/SwiftExpr/Dict.test
Comment (vedantk) [inline lit/SwiftExpr/Dict.test]: Now that we're using FileCheck, we can tighten these checks up with -NEXT. But it'll do as a follow-up.

**Review by vedantk — APPROVED** [review]
LGTM, thanks!

**Comment by dcci** [comment]
I decided to be a little more conservative in this. If you think it's always safe also in the repl to use `CHECK-NEXT` I'll go for it.

**Comment by vedantk** [comment]
If there's repl output between those check lines, I'd be concerned / would want a test failure to learn about the issue. I don't feel very strongly about this and wouldn't block the PR over it.

**Comment by dcci** [comment]
Also, thanks to @jrose-apple for suggesting the right way of testing multiple commands via `<` :)

**Comment by jrose-apple** [comment]
Uh, note that you should use `//` for your comments, then! `#` is not a comment character in Swift.

**Comment by dcci** [comment]
@jrose-apple thanks! I'll update my pull request accordingly.

**Review by JDevlieghere — COMMENTED** [review]
Great, thanks for taking the time to improve these tests! 
LGTM (modulo comment fix)

**Comment by dcci** [comment]
@swift-ci please test

**Comment by dcci** [comment]
@swift-ci please test

**Pull request merged**
